File: sdk/chocobase-go/postgrest.go
package chocobase

import "fmt"

// PostgrestClient manages database queries.
type PostgrestClient struct {
	BaseURL string
	APIKey  string
}

func newPostgrestClient(baseURL, apiKey string) *PostgrestClient {
	return &PostgrestClient{
		BaseURL: baseURL,
		APIKey:  apiKey,
	}
}

// From returns a QueryBuilder for the specified table.
func (p *PostgrestClient) From(table string) *QueryBuilder {
	return &QueryBuilder{
		BaseURL: fmt.Sprintf("%s/rest/v1/%s", p.BaseURL, table),
		Table:   table,
		APIKey:  p.APIKey,
		Params:  make(map[string]string),
	}
}

// QueryBuilder fluently constructs PostgREST queries.
type QueryBuilder struct {
	BaseURL string
	Table   string
	APIKey  string
	Params  map[string]string
}

func (q *QueryBuilder) Select(columns string) *QueryBuilder {
	q.Params["select"] = columns
	return q
}

func (q *QueryBuilder) Eq(column string, value interface{}) *QueryBuilder {
	q.Params[column] = fmt.Sprintf("eq.%v", value)
	return q
}

func (q *QueryBuilder) Neq(column string, value interface{}) *QueryBuilder {
	q.Params[column] = fmt.Sprintf("neq.%v", value)
	return q
}

func (q *QueryBuilder) Gt(column string, value interface{}) *QueryBuilder {
	q.Params[column] = fmt.Sprintf("gt.%v", value)
	return q
}

func (q *QueryBuilder) Lt(column string, value interface{}) *QueryBuilder {
	q.Params[column] = fmt.Sprintf("lt.%v", value)
	return q
}

func (q *QueryBuilder) Limit(count int) *QueryBuilder {
	q.Params["limit"] = fmt.Sprintf("%d", count)
	return q
}

// QueryResult represents standard query execution payload.
type QueryResult struct {
	Data  []map[string]interface{} `json:"data"`
	Count int                      `json:"count"`
	Error string                   `json:"error,omitempty"`
}

func (q *QueryBuilder) Execute() (*QueryResult, error) {
	return &QueryResult{
		Data:  make([]map[string]interface{}, 0),
		Count: 0,
	}, nil
}

File: sdk/chocobase-rs/src/postgrest.rs
use std::collections::HashMap;

#[derive(Debug, Clone)]
pub struct QueryBuilder {
    pub base_url: String,
    pub table: String,
    pub api_key: String,
    pub params: HashMap<String, String>,
}

impl QueryBuilder {
    pub fn select(mut self, columns: &str) -> Self {
        self.params.insert("select".to_string(), columns.to_string());
        self
    }

    pub fn eq(mut self, column: &str, value: impl ToString) -> Self {
        self.params.insert(column.to_string(), format!("eq.{}", value.to_string()));
        self
    }

    pub fn neq(mut self, column: &str, value: impl ToString) -> Self {
        self.params.insert(column.to_string(), format!("neq.{}", value.to_string()));
        self
    }

    pub fn gt(mut self, column: &str, value: impl ToString) -> Self {
        self.params.insert(column.to_string(), format!("gt.{}", value.to_string()));
        self
    }

    pub fn lt(mut self, column: &str, value: impl ToString) -> Self {
        self.params.insert(column.to_string(), format!("lt.{}", value.to_string()));
        self
    }

    pub fn limit(mut self, count: usize) -> Self {
        self.params.insert("limit".to_string(), count.to_string());
        self
    }

    pub fn execute(&self) -> serde_json::Value {
        serde_json::json!({
            "data": [],
            "count": 0,
            "error": null
        })
    }
}

#[derive(Debug, Clone)]
pub struct PostgrestClient {
    pub base_url: String,
    pub api_key: String,
}

impl PostgrestClient {
    pub fn new(base_url: String, api_key: String) -> Self {
        Self { base_url, api_key }
    }

    pub fn from(&self, table: &str) -> QueryBuilder {
        QueryBuilder {
            base_url: format!("{}/rest/v1/{table}", self.base_url),
            table: table.to_string(),
            api_key: self.api_key.clone(),
            params: HashMap::new(),
        }
    }
}
